notebooks/02_model/vowpal_wabbit_deep_dive.ipynb

```diff
@@ -0,0 +1,737 @@
+    "        print('{name}: {metric}'.format(name=f.__name__.upper(), metric=f(test_data, pred_data)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create temp directory to maintain data files\n",
+    "tmpdir = TemporaryDirectory()\n",
+    "\n",
+    "model_path = os.path.join(tmpdir.name, 'vw.model')\n",
+    "train_path = os.path.join(tmpdir.name, 'train.dat')\n",
+    "test_path = os.path.join(tmpdir.name, 'test.dat')\n",
+    "train_logistic_path = os.path.join(tmpdir.name, 'train_logistic.dat')\n",
+    "test_logistic_path = os.path.join(tmpdir.name, 'test_logistic.dat')\n",
+    "prediction_path = os.path.join(tmpdir.name, 'prediction.dat')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h3>Load & Transform Data</h3>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# load movielens data (use the 1M dataset)\n",
+    "df = load_pandas_df('1m')\n",
+    "\n",
+    "# split data to train and test sets, default values take 75% of each users ratings as train, and 25% as test\n",
+    "train, test = python_stratified_split(df)\n",
+    "\n",
+    "# save train and test data in vw format\n",
+    "to_vw(df=train, output=train_path)\n",
+    "to_vw(df=test, output=test_path)\n",
+    "\n",
+    "# save data for logistic regression (requires adjusting the label)\n",
+    "to_vw(df=train, output=train_logistic_path, logistic=True)\n",
+    "to_vw(df=test, output=test_logistic_path, logistic=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h3>Regression Based Recommendations</h3>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When considering different approaches for solving a problem with machine learning it is helpful to generate a baseline approach to understand how more complex solutions perform across dimensions of performance, time, and resource (memory or cpu) usage.\n",
+    "\n",
+    "One of the most basic approaches for a recommendation engine is to simply learn a linear regression model that is trained on examples of ratings as the target variable and corresponding user ids and movie ids as independent features.\n",
+    "\n",
+    "By passing each user-item rating in as an example the model will begin to learn weights based on average ratings for each user as well as average ratings per item.\n",
+    "\n",
+    "VW uses linear regression by default, so no extra command line options are needed beyond specifying where to locate the model and the data.\n",
+    "\n",
+    "This however generates predicted ratings which are no longer integers, so some additional adjustments can be made at prediction time to convert them back to the integer scale of 1 through 5 if necessary. Here this is done in the evaluate function."
```

Comment:
are you asking if the implementation in this notebook supports tuning? If so, no it is not really set up for that. However, it is possible to modify the run_vw wrapper to support such behavior, you would just need to add additional input to inject hyperparameters and format them in the input string correctly.
Although it would be easier to use the python bindings for this, which allows for the same kind of parameter sweeping that can be done with sk-learn models.